File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        
        int max = 0 ;
        for(int i=0 ; i < s.Length ; i++){
            HashSet<char> ans = new HashSet<char>();

            
            for(int j=i ; j < s.Length ; j++){
                if(ans.Contains(s[j])) break;

                ans.Add(s[j]);

                if(ans.Count > max) max = ans.Count;
            }
        }

        return max ;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_set<char> st ;

        int j = 0 ;
        int maxlen = 0 ;
        for( int i=0 ; i<s.length() ; i++){
           
           while( st.find( s[i] ) != st.end() ){
                st.erase(s[j]);
                j++;
           }

           st.insert( s[i] );
           maxlen = max ( maxlen , i-j+1);
        }
        return maxlen ;
    }
};